### PR TITLE
Update docs for caching, test writing and pitfalls

### PR DESCRIPTION
**Pull request by foursixnine**
Add documentation for the needle caching, now that it has been
deployed, it is useful to have it documented.

Add a pitfall i run myself into every time i update my openQA packages
(which is probably a bad practice).

### CODE REVIEW

File: docs/Installing.asciidoc [hunk 1]
```diff
@@ -311,6 +310,22 @@ NFS clients are where openQA workers are running. Run following command:
 mount -t nfs openQA-webUI-host:/var/lib/openqa/share /var/lib/openqa/share
 --------------------------------------------------------------------------------
 
+Needle Caching  
+~~~~~~~~~~~~~~~
+
+If you your network is slow or you experience long time to load needles, you
+might want to consider needle caching, to do this a directory
++/var/lib/openqa/cache+ must be created, and right permissions given to the
+'geekotest' user.
```
Comment (okurz) [inline docs/Installing.asciidoc]: If your network is slow or you experience long loading time for needles you
might want to consider needle caching. To use needle caching a directory
++/var/lib/openqa/cache+ must be created and and right permissions must given to the
+'geekotest' user.

File: docs/Pitfalls.asciidoc [hunk 2]
```diff
@@ -7,10 +7,25 @@ openQA pitfalls
...
   will not be listed in old tests.
-- if an existing needle is updated with a new image or different
+- If an existing needle is updated with a new image or different
   areas, the old test will display the new needle which might be
   confusing
-- if a needle is deleted, old tests may display an error when viewing
+- If a needle is deleted, old tests may display an error when viewing
   them in the web UI.
+
+Mixed production and Development environment
```
Comment (okurz) [inline docs/Pitfalls.asciidoc]: s/Development/development/

File: docs/Pitfalls.asciidoc [hunk 3]
```diff
@@ -7,10 +7,25 @@ openQA pitfalls
...
+a packaged version of openqa:
+
+If the setup for the development scenario involves sharing /var/lib/openqa,
+it would be wise to have a shared group _openqa_, that will have write and execute
+permissions over said directory, so that _geekotest_ user and the normal development
+user can share the environment without problems.
+
+This approach will lead to a problem when the openqa package is updated, since the
+directory permissions will be changed again, nothing a `chmod -R g+rwx /var/lib/openqa/`
+and `chgrp -R openqa /var/lib/openqa` can't fix.
```
Comment (okurz) [inline docs/Pitfalls.asciidoc]: s/can't/can not/

File: docs/Installing.asciidoc [hunk 4]
```diff
@@ -311,6 +310,23 @@ NFS clients are where openQA workers are running. Run following command:
 mount -t nfs openQA-webUI-host:/var/lib/openqa/share /var/lib/openqa/share
 --------------------------------------------------------------------------------
 
+Needle Caching  
+~~~~~~~~~~~~~~~
+
+If you your network is slow or you experience long time to load needles you
```
Comment (aaannz) [inline docs/Installing.asciidoc]: There is still double `you`

Comment (foursixnine) [inline docs/Installing.asciidoc]: Well the double you is there for a reason :), a comma might help... 

> If you your network is slow or you experience long time to load needles, you might want to consider needle caching.
We can also use:
> If you your network is slow or needles are taking too long to load, you might want to consider needle caching.

Comment (aaannz) [inline docs/Installing.asciidoc]: That doesn't make sense to me. What are you trying to say?

Comment (foursixnine) [inline docs/Installing.asciidoc]: That the double you even if it makes the sentence sound a bit odd, is trying to give a hint to the user on what to do, I can't think of a better way to offer the solution than the second sentence i wrote.

Comment (foursixnine) [inline docs/Installing.asciidoc]: @aaannz, sorry my bad here... you mean the _If you your..._ right? (somehow i skipped this when proofreading :P) - fixed -

File: docs/Pitfalls.asciidoc [hunk 5]
```diff
@@ -7,10 +7,25 @@ openQA pitfalls
...
+- If a needle is deleted, old tests may display an error when viewing
   them in the web UI.
+
+Mixed production and development environment
+============================================
+
+There are few things to take into account when running a development version and
+a packaged version of openqa:
+
+If the setup for the development scenario involves sharing /var/lib/openqa,
```
Comment (okurz) [inline docs/Pitfalls.asciidoc]: can you put the path argument in `+`? -> `+/var/lib/openqa+`

File: docs/Installing.asciidoc [hunk 6]
```diff
@@ -311,6 +310,23 @@ NFS clients are where openQA workers are running. Run following command:
 mount -t nfs openQA-webUI-host:/var/lib/openqa/share /var/lib/openqa/share
 --------------------------------------------------------------------------------
 
+Needle Caching  
+~~~~~~~~~~~~~~~
+
+If your network is slow or you experience long time to load needles you
+might want to consider needle caching. To use needle caching a directory
```
Comment (okurz) [inline docs/Installing.asciidoc]: then make it show them correctly :-) Are you sure that this code differs from what you have locally?

File: docs/Pitfalls.asciidoc (hunk 5 above)
Comment (foursixnine) [inline docs/Pitfalls.asciidoc]: Done

File: docs/Installing.asciidoc (hunk 6 above)
Comment (foursixnine) [inline docs/Installing.asciidoc]: It should display correctly now... idk what happened, but shouldn't be a problem now

**Comment by aaannz** [comment]
:+1:

**Pull request merged**